### ISSUE TITLE
Why there is nn.softmax before nn.crossentroyloss in infogan?

### ISSUE THREAD

**Issue by Lucksong** [eriklindernoren/PyTorch-GAN#84]
In 111 line of the implementaion of infogan, nn.softmax is added to aux_layer. But the returned data of aux_layer is used to compute crossentropy loss in 277 line. Doesn`t  nn.crossentropyloss  module 
 integrated  functions of log_softmax and  nllloss?

**Comment by Lucksong** [comment]
One more thing, sample_image() function is for verification, shouldn`t we set the model to test mode firstly?

**Comment by mo666666** [comment]
> In 111 line of the implementaion of infogan, nn.softmax is added to aux_layer. But the returned data of aux_layer is used to compute crossentropy loss in 277 line. Doesn`t  nn.crossentropyloss  module
> integrated  functions of log_softmax and  nllloss?

I also think nn.softmax is not needed here.
